lightwood/api/data_source.py

```diff
@@ -60,6 +60,8 @@ def __init__(self, data_frame, configuration):
         self.enable_dropout = False
         self.disable_cache = not CONFIG.CACHE_ENCODED_DATA
         self.subsets = {}
+        self.out_indexes = None
+        self.out_types = None
```

Comment:
I can't get the output indexes (`out_indexes`) because those are only determined once we transform the data.

I can get the output type (`out_types`) out of the config... but, it's rather annoying to get to them and I'm relying on the ordering here (as in, the output indexes correspond to the types in order), and that's not decided in the config, that's decided in the transform. So I think it's better to source them this way. 

Overall, this part is a bit messy, but all other solutions were messy as well so I ended up going with this approach.